src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -120,7 +120,13 @@ macro_rules! word_cnt {
         word_cnt!($len, u8)
     };
     ($len:expr, $t:ty) => {{
-        (($len) + DIGITS_PER_WORD as $t - 1) / DIGITS_PER_WORD as $t
+        let len = ($len) as $t;
+        if len > (DIGITS_PER_WORD * WORD_BUF_LEN) as $t {
+            // process overflow
+            (WORD_BUF_LEN + 1) as $t
```

Comment:
How about `cmp::max((($len) + DIGITS_PER_WORD as $t - 1) / DIGITS_PER_WORD as $t, WORD_BUF_LEN + 1)`?

Reply:
no, since `($len) + DIGITS_PER_WORD as $t ` maybe overflow in some cases.